gateway/repository.go

```diff
@@ -0,0 +1,55 @@
+package gateway
+
+import (
+	"context"
+	"github.com/TykTechnologies/tyk/config"
+	"github.com/TykTechnologies/tyk/user"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type Database interface {
+	CreatePolicy(ctx context.Context, policy *user.Policy) error
+}
+
+type DB struct {
+	*mongo.Database
+}
```

Comment:
I'd call it PolicyDB or AnalyticsDB and there is no need for pointer.

Reply:
nope, there is neep pointer
DB was renamed to `AnalyticsDB`